packages/loot-core/src/client/data-hooks/filters.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react';
+
+import q, { liveQuery } from '../query-helpers';
+
+function toJS(rows) {
+  let filters = rows.map(row => {
+    return {
+      ...row.fields,
+      id: row.id,
+      name: row.name,
+      tombstone: row.tombstone,
+      conditionsOp: row.conditions_op,
+      conditions: row.conditions,
+    };
+  });
+  return filters;
+}
+
+export function useFilters() {
+  let [data, setData] = useState([]);
+
+  useEffect(() => {
+    let query = q('transaction_filters').select('*');
+
+    let filterQuery = liveQuery(query, async filters => {
+      let filte = toJS(filters);
+      setData(filte);
+    });
+
+    return () => {
+      filterQuery.unsubscribe();
+    };
+  }, []);
+
+  return data;
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏you should be able to use `useLiveQuery` for this + `toJS` mutation of the returned data.

Reply:
It keeps coming back null for me. What am I doing wrong?


```
  let fil = useLiveQuery(() => q('transaction_filters').select('*'), [])
  let filte = toJS(fil);
```

Reply:
It will be `null` for no results. So you might want to do this to force a different empty state:

```
  let fil = useLiveQuery(() => q('transaction_filters').select('*'), [])
  let filte = toJS(fil || []);
```

Reply:
Worked great, thanks!